Build/Build.cs

```diff
@@ -97,19 +102,32 @@ public override ISbomGeneratorBuilder ConfigureSbomGenerationInfoBuilder (Soluti
 
   public override void ConfigureProjects (ProjectsBuilder projects)
   {
-    var normalTestConfiguration = new TestConfiguration(
-        DefaultTestExecutionRuntimeFactory.Instance,
-        TestMatrices.Single(e => e.Name == "NormalTestMatrix"),
+    [CanBeNull]
+    static TestConfiguration CreateTestConfiguration (
+        [CanBeNull] TestMatrix testMatrix,
+        ITestExecutionRuntimeFactory testExecutionRuntimeFactory,
+        ImmutableArray<ITestExecutionWrapper> testExecutionWrappers)
+    {
+      return testMatrix != null
+          ? new TestConfiguration(testExecutionRuntimeFactory, testMatrix, testExecutionWrappers)
+          : null;
+    }
+
+    var testExecutionRuntimeFactory = new DefaultTestExecutionRuntimeFactory(new DockerNetworkDockerRunSettingsCustomizer());
+
+    var normalTestConfiguration = CreateTestConfiguration(
+        _normalTestMatrix,
+        testExecutionRuntimeFactory,
         ImmutableArray<ITestExecutionWrapper>.Empty);
 
-    var webTestingTestConfiguration = new TestConfiguration(
-        DefaultTestExecutionRuntimeFactory.Instance,
-        TestMatrices.Single(e => e.Name == "WebTestingTestMatrix"),
+    var webTestingTestConfiguration = CreateTestConfiguration(
+        _webTestMatrix,
+        testExecutionRuntimeFactory,
         [new WebTestingTestSetup()]);
 
-    var databaseTestConfiguration = new TestConfiguration(
-        DefaultTestExecutionRuntimeFactory.Instance,
-        TestMatrices.Single(e => e.Name == "DatabaseTestMatrix"),
+    var databaseTestConfiguration = CreateTestConfiguration(
+        _databaseTestMatrix,
+        testExecutionRuntimeFactory,
         [new DatabaseTestSetup()]);
```

Comment:
shouldn't guard / give a meaningful error if there is an incorrect method call order that leaves the _matrix null?

Reply:
So the whole thing with nullable `TestMatrix` here is that we want to support partial builds. When tests are executed, the test matrix can be calculated and the test matrices are non-null. But if we don't do a test build we have no matrices but we still might want to get project metadata so we need this method to work even if there is no test matrix.

Reply:
okay, misunderstanding: I was assuming that CreateTestConfig guards this null-safely :D
Maybe a comment would be helpful here because that's an implicit detail that might get lost down the road.